tests/test_udunits.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+import xwrf
+
+from . import importorskip
+
+
+@pytest.fixture(scope='session')
+def sample_dataset(request):
+    return xwrf.tutorial.open_dataset(request.param)
+
+
+@importorskip('cf_units')
+@pytest.mark.parametrize('sample_dataset', ['lambert_conformal'], indirect=True)
+@pytest.mark.parametrize(
+    'variable,cf_definition_string',
+    (
+        ('QNRAIN', 'kg-1'),
+        ('NOAHRES', 'kg.s-3'),
+        ('THIS_IS_AN_IDEAL_RUN', '?'),
+        ('SAVE_TOPO_FROM_REAL', '?'),
+        ('OL4', '?'),
+        ('VAR_SSO', 'm2'),
+        ('SEAICE', '?'),
+    ),
+)
+def test_problematic_units_against_udunits_wrfout(sample_dataset, variable, cf_definition_string):
+    from cf_units import Unit
+
+    ds = sample_dataset.xwrf.postprocess()
+    assert Unit(ds[variable].attrs.get('units', '')).definition == cf_definition_string
+
+
+@importorskip('cf_units')
+@pytest.mark.parametrize('sample_dataset', ['met_em_sample'], indirect=True)
+@pytest.mark.parametrize(
+    'variable,cf_definition_string',
+    (
+        ('OL4', '1'),
```

Comment:
Why is this different to L22? Do we want it to be different?

Reply:
I believe that's just a difference in the source files...`'met_em_sample'` has:

```
float OL4(Time, south_north, west_east) ;
        OL4:FieldType = 104 ;
        OL4:MemoryOrder = "XY " ;
        OL4:units = "fraction" ;
        OL4:description = "effective orographic length" ;
        OL4:stagger = "M" ;
        OL4:sr_x = 1 ;
        OL4:sr_y = 1 ;
```

whereas `'lambert_conformal'` has:

```
float OL4(Time, south_north, west_east) ;
        OL4:FieldType = 104 ;
        OL4:MemoryOrder = "XY " ;
        OL4:description = "OROGRAPHIC DIRECTION ASYMMETRY FUNCTION" ;
        OL4:units = "" ;
        OL4:stagger = "" ;
        OL4:coordinates = "XLONG XLAT XTIME" ;
```